ts/packages/anchor/src/coder/borsh/idl.ts

```diff
@@ -123,45 +124,51 @@ export class IdlCoder {
     types: IdlTypeDef[] = [],
     name?: string
   ): Layout {
-    if (typeDef.type.kind === "struct") {
-      const fieldLayouts = typeDef.type.fields.map((field) => {
-        const x = IdlCoder.fieldLayout(field, types);
-        return x;
-      });
-      return borsh.struct(fieldLayouts, name);
-    } else if (typeDef.type.kind === "enum") {
-      let variants = typeDef.type.variants.map((variant: IdlEnumVariant) => {
-        const name = camelCase(variant.name);
-        if (variant.fields === undefined) {
-          return borsh.struct([], name);
-        }
-        const fieldLayouts = variant.fields.map(
-          (f: IdlField | IdlType, i: number) => {
-            if (!f.hasOwnProperty("name")) {
+    switch (typeDef.type.kind) {
+      case "struct": {
+        const fieldLayouts = typeDef.type.fields.map((field) => {
+          return IdlCoder.fieldLayout(field, types);
+        });
+        return borsh.struct(fieldLayouts, name);
+      }
+
+      case "enum": {
+        let variants = typeDef.type.variants.map((variant: IdlEnumVariant) => {
+          const name = camelCase(variant.name);
+          if (!variant.fields) {
+            return borsh.struct([], name);
+          }
+
+          const fieldLayouts = variant.fields.map(
+            (f: IdlField | IdlType, i: number) => {
+              if ((f as IdlField)?.name) {
+                return IdlCoder.fieldLayout(f as IdlField, types);
+              }
+
               return IdlCoder.fieldLayout(
                 { type: f as IdlType, name: i.toString() },
                 types
               );
             }
-            // this typescript conversion is ok
-            // because if f were of type IdlType
-            // (that does not have a name property)
-            // the check before would've errored
-            return IdlCoder.fieldLayout(f as IdlField, types);
-          }
-        );
-        return borsh.struct(fieldLayouts, name);
-      });
+          );
+          return borsh.struct(fieldLayouts, name);
+        });
 
-      if (name !== undefined) {
-        // Buffer-layout lib requires the name to be null (on construction)
-        // when used as a field.
-        return borsh.rustEnum(variants).replicate(name);
+        if (name !== undefined) {
+          // Buffer-layout lib requires the name to be null (on construction)
+          // when used as a field.
+          return borsh.rustEnum(variants).replicate(name);
+        }
+
+        return borsh.rustEnum(variants, name);
       }
 
-      return borsh.rustEnum(variants, name);
-    } else {
-      throw new Error(`Unknown type kint: ${typeDef}`);
+      case "alias": {
+        return IdlCoder.fieldLayout(
+          { type: typeDef.type.value, name: typeDef.name },
+          types
+        );
+      }
```

Comment:
Looks like you dropped the catch-all `else` but didn't include a `default` case? ie:

```typescript
default: {
   throw new Error(`Unknown type kind: ${typeDef}`);
}
```

The original is also spelled wrong lol (`type kint`)

Reply:
Yes, I've dropped it because it should be unreachable, i.e. the value we are checking is generated from the IDL so it should never be outside of these specific cases.